backend/amd64/proc.ml

```diff
@@ -459,6 +460,37 @@ let destroyed_at_terminator (terminator : Cfg_intf.S.terminator) =
     Misc.fatal_error "no instructions specific for this architecture can raise"
   | Poll_and_jump _ -> destroyed_at_alloc_or_poll
 
+(* CR-soon xclerc for xclerc: consider having more destruction points.
+   We current return `true` when `destroyed_at_terminator` returns
+   `all_phys_regs`; we could also return `true` when `destroyed_at_terminator`
+   returns `destroyed_at_c_call` for instance. *)
+(* note: keep this function in sync with `destroyed_at_terminator` above. *)
+let is_destruction_point (terminator : Cfg_intf.S.terminator) =
```

Comment:
I don't think it would be proper to have
a destruction point as soon as registers
are destroyed, because that would (on
`amd64`) for instance include `Switch`
which destroys only 2 registers.

That being said, I experimented with
what the `CR-soon` immediately above
suggests, and the results are exactly
one would expect:
- spill a bit more;
- save some IRC rounds.

I am note sure yet which is better overall,
as we are ~trading build time for execution
time.